### PR TITLE
Add ConsistencyEvaluator with BERT and ROUGE-based similarity metrics

### PR DESCRIPTION
**Pull request by vish119**
**Consistency Evaluator PR 2**
This PR introduces a new ConsistencyEvaluator that calculates consistency scores across a set of generated responses for a given query using BERTScore and ROUGE-L. Unlike traditional evaluation metrics that assess a single response, this evaluator measures the internal alignment or consistency among multiple generated answers for the same query. It also includes a plotting utility to visualize distributional statistics such as mean, median, standard deviation, and interquartile range (IQR) for each consistency metric.

### Structural Changes

- The ConsistencyEvaluator returns a `ConsistencyResult` object per query. This object contains:
  - A `MultiRAGResult`, representing the generated responses being evaluated for consistency.
  - A dictionary of `ConsistencyScore` objects (one per metric like BERT or ROUGE), each storing raw consistency values and statistical summaries.

Additionally, the ConsistencyEvaluator includes functionality to plot consistency metrics across queries to support exploratory analysis.

### Output Behavior

- Evaluation results are saved in dedicated CSV files for each evaluator (e.g., TREC and ConsistencyEvaluator).
- A consolidated CSV file combining results from all evaluators is automatically generated.

These changes lay the groundwork for incorporating consistency-based evaluation into multi-response RAG pipelines, offering deeper insight into the stability and alignment of generated outputs.

### CODE REVIEW

File: open_rag_eval/metrics/bert_answer_similarity_metric.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from typing import List
+
+from sklearn.metrics.pairwise import cosine_similarity
+from sentence_transformers import SentenceTransformer
+
+from open_rag_eval.data_classes.rag_results import MultiRAGResult
+from open_rag_eval.metrics.base_metrics import PairwiseAnswerSimilarityMetric
+
+
+class BERTAnswerSimilarityMetric(PairwiseAnswerSimilarityMetric):
```
Comment (ofermend) [inline open_rag_eval/metrics/bert_answer_similarity_metric.py:12]: Which languages does BERTAnswerSimilarityMetric work with? We may want to add this to the README. Same for ROGUE based metric.
